src/services/jobRoleApiService.ts

```diff
@@ -66,7 +66,8 @@ export class JobRoleApiService implements JobRoleservice {
       const response = await axios.get<ApiResponse<JobRole[]>>(
         `${this.baseURL}/jobs`
       );
-      return response.data.data || [];
+      const jobs = response.data.data || [];
+      return jobs.map((job) => this.mapJobData(job));
     } catch (error) {
       console.error("Error fetching jobs from API:", error);
```

Comment:
This error message was good - but the behaviour has changed a little bit now. 

We can now have a failure where Jobs are present from `const jobs = response.data.data || [];` 

but `jobs.map((job) => this.mapJobData(job));` could throw an error. 

We should have a look at making this error better or maybe handling it in mapJobData so it better reflects what can happen